src/Domain/JacobianDiagnostic.cpp

```diff
@@ -15,48 +15,64 @@
 namespace domain {
 template <size_t Dim, typename Fr>
 void jacobian_diagnostic(
-    const gsl::not_null<Scalar<DataVector>*> jacobian_diagnostic,
+    const gsl::not_null<
+        tnsr::i<DataVector, Dim, typename Frame::ElementLogical>*>
+        jacobian_diag,
     const Jacobian<DataVector, Dim, typename Frame::ElementLogical, Fr>&
         analytic_jacobian,
     const TensorMetafunctions::prepend_spatial_index<
         tnsr::I<DataVector, Dim, Fr>, Dim, UpLo::Lo,
         typename Frame::ElementLogical>& numeric_jacobian_transpose) {
-  auto jacobian_difference_sum =
-      make_with_value<tnsr::i<DataVector, Dim, typename Frame::ElementLogical>>(
-          get<0, 0>(analytic_jacobian), 0.0);
-  auto numerical_difference_sum =
-      make_with_value<tnsr::i<DataVector, Dim, typename Frame::ElementLogical>>(
-          get<0, 0>(analytic_jacobian), 0.0);
+  destructive_resize_components(jacobian_diag,
+                                analytic_jacobian.begin()->size());
+  const size_t number_of_points = analytic_jacobian.begin()->size();
+  auto analytic_jacobian_sum =
+      tnsr::i<DataVector, Dim, typename Frame::ElementLogical>{number_of_points,
+                                                               0.0};
+  auto numerical_jacobian_sum =
+      tnsr::i<DataVector, Dim, typename Frame::ElementLogical>{number_of_points,
+                                                               0.0};
   constexpr double eps_to_avoid_div_by_zero = 1.e-20;
 
   // i_hat = logical frame index
   // i = mapped frame index
   for (size_t i_hat = 0; i_hat < Dim; ++i_hat) {
     for (size_t i = 0; i < Dim; ++i) {
-      jacobian_difference_sum.get(i_hat) -= analytic_jacobian.get(i, i_hat);
-      numerical_difference_sum.get(i_hat) +=
-          numeric_jacobian_transpose.get(i_hat, i);
+      analytic_jacobian_sum.get(i_hat) +=
+          square(analytic_jacobian.get(i, i_hat));
+      numerical_jacobian_sum.get(i_hat) +=
+          square(numeric_jacobian_transpose.get(i_hat, i));
     }
   }
   for (size_t i_hat = 0; i_hat < Dim; ++i_hat) {
-    // If the numeric and analytic components are both sufficiently small,
-    // treat them as equal.
-    constexpr double eps_for_ratio = 1.e-10;
-    if (max(abs(jacobian_difference_sum.get(i_hat))) < eps_for_ratio and
-        max(abs(numerical_difference_sum.get(i_hat))) < eps_for_ratio) {
-      jacobian_difference_sum.get(i_hat) = 0.0;
-      continue;
-    }
-    jacobian_difference_sum.get(i_hat) /=
-        (numerical_difference_sum.get(i_hat) + eps_to_avoid_div_by_zero);
-    jacobian_difference_sum.get(i_hat) += 1.0;
+    analytic_jacobian_sum.get(i_hat) =
+        sqrt(analytic_jacobian_sum.get(i_hat)) * -1.0;
+    analytic_jacobian_sum.get(i_hat) /=
+        (sqrt(numerical_jacobian_sum.get(i_hat)) + eps_to_avoid_div_by_zero);
```

Comment:
Maybe instead of the epsilon thing you just set the quantity to NaN (or `quiet_NaN`) if the denominator is zero? That should propagate fine to ParaView.

Reply:
I prefer leaving the epsilon as-is

Reply:
so, delete the eps?